### PR TITLE
Optional `scripts` section in libraries' config

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

This change adds a scripts/hooks section to config that allows to specify a list of available scripts for the given library. Scripts are defined in a free-form, meaning that any kind of value can be provided as a value and is parsed as a string.

For example,
```
scripts:
  refresh:
    - Standard.Base.HTTP.Caches.refresh
```

Part of #11485.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
There is a checklist in this PR (as well as in any other PR) asking for few checks including:

>  - [ ] The documentation has been updated, if necessary.

I guess this time, when we design a **new API** for library writers, we should start by **writing the documentation**, updating changelog. Etc.

> adds a scripts/hooks section to config that allows to specify a list of available scripts for the given library. Scripts are defined in a free-form, meaning that any kind of value can be provided as a value and is parsed as a string.

I'd like to see list of use-cases justifying the API decisions first. Really, "freeform"? Why? What are the scripts good for? How can one use them? When they are called?

**Comment by hubertp** [comment]
> I'd like to see list of use-cases justifying the API decisions first.

This PR unblocks further work that we have discussed internally for the purpose of #11485. Thanks for your feedback.

File: engine/runner/src/main/java/org/enso/runner/Main.java [hunk 1]
```diff
@@ -599,7 +600,8 @@ private void createNew(
             authors,
             nil(),
             "",
-            Option$.MODULE$.empty());
+            Option$.MODULE$.empty(),
+            List$.MODULE$.empty());
```
Comment (4e6) [inline engine/runner/src/main/java/org/enso/runner/Main.java:604]: I think you can use `nil()` helper method (see two lines above)

File: lib/scala/pkg/src/main/scala/org/enso/pkg/Script.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    if (knownScripts.contains(name)) Right(Script(name, values))
+    else Left("Uknown script: " + name)
+  }
+
+  /** [[Decoder]] instance for the [[Script]]. */
+  implicit val decoder: Decoder[Script] = { json =>
+    for {
+      key    <- json.key.toRight(DecodingFailure("no key", Nil))
+      fields <- json.get[List[String]](key)
+      script <- find(key, fields).left.map(v => DecodingFailure(v, Nil))
```
Comment (4e6) [inline lib/scala/pkg/src/main/scala/org/enso/pkg/Script.scala]: We should not fail the whole `scripts` section if one element fails. It is bad for backward compatibility. In general, we should allow arbitrary config structure and ignore everything irrelevant.

File: lib/scala/pkg/src/main/scala/org/enso/pkg/Script.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.yaml.snakeyaml.error.YAMLException
+import org.yaml.snakeyaml.nodes.{MappingNode, Node, ScalarNode, SequenceNode}
+
+import java.util
+
+case class Script(name: String, arguments: Seq[String])
+
+object Script {
+
+  val knownScripts = List("refresh")
```
Comment (4e6) [inline lib/scala/pkg/src/main/scala/org/enso/pkg/Script.scala]: I'd parse everything. The logic will make use of relevant parts

**Comment by 4e6** [comment]
Please update `ConfigSpec`

File: lib/scala/pkg/src/main/scala/org/enso/pkg/Script.scala (hunk 2 above)
Comment (hubertp) [reply lib/scala/pkg/src/main/scala/org/enso/pkg/Script.scala]: This is a bit random as we sometimes fail and sometimes not, when parsing Config. But if I remove the known scripts check then that will resolve itself.

**Review by 4e6 — APPROVED** [review]

**Comment by hubertp** [comment]
Currently on hold to test an alternative solution https://github.com/enso-org/enso/issues/11485#issuecomment-2478082470

**Comment by hubertp** [comment]
Made obsolete via https://github.com/enso-org/enso/pull/11577. We could still revive it, if needed.

**Pull request closed without merging**